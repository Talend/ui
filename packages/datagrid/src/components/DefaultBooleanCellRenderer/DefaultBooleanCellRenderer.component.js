import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import DefaultValueRenderer from '../DefaultCellRenderer/DefaultValueRenderer.component';

export default function DefaultBooleanCellRenderer({ data }) {
	// convert the boolean to string to render this value
	const value = String(data.value);
	return <DefaultValueRenderer value={value} className={classNames('td-cell-boolean')} />;
}

DefaultBooleanCellRenderer.propTypes = {
	data: PropTypes.shape({
		value: PropTypes.oneOfType([PropTypes.string, PropTypes.bool, PropTypes.number]),
	}),
};
