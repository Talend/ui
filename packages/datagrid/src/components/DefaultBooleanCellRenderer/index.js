import DefaultBooleanCellRenderer from './DefaultBooleanCellRenderer.component';

export default DefaultBooleanCellRenderer;
