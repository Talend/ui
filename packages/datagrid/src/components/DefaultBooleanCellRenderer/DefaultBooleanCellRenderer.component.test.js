import React from 'react';
import { shallow } from 'enzyme';

import DefaultBooleanCellRenderer from './DefaultBooleanCellRenderer.component';

describe('DatagridBooleanCellRenderer', () => {
	it('should render a boolean value', () => {
		const wrapper = shallow(<DefaultBooleanCellRenderer data={{ value: false }} />);

		expect(wrapper.getElement()).toMatchSnapshot();
	});

	it('should render a int value', () => {
		const wrapper = shallow(<DefaultBooleanCellRenderer data={{ value: 0 }} />);

		expect(wrapper.props().value).toBe('0');
		expect(wrapper.find('DefaultValueRenderer').length).toBe(1);
	});

	it('should render a string value', () => {
		const wrapper = shallow(<DefaultBooleanCellRenderer data={{ value: 'value' }} />);

		expect(wrapper.props().value).toBe('value');
		expect(wrapper.find('DefaultValueRenderer').length).toBe(1);
	});
});
