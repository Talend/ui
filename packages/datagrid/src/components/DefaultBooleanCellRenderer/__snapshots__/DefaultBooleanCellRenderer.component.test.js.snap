// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatagridBooleanCellRenderer should render a boolean value 1`] = `
<DefaultValueRenderer
  className="td-cell-boolean"
  value="false"
/>
`;
